src/elements/play-assets/play-assets.ts

```diff
@@ -0,0 +1,429 @@
+import {type PropertyValues, ReactiveElement} from 'lit'
+import {customElement, property} from 'lit/decorators.js'
+import {WebAccess, WebStorage} from '@zenfs/dom'
+import {Bubble} from '../../utils/bubble.js'
+import {Zip} from '@zenfs/zip'
+import {
+  hasFileAccessAPI,
+  tryGetFile,
+  tryQueryPermission
+} from '../../utils/file-access-api.js'
+import * as ZenFS from '@zenfs/core'
+import {FileSystem, fs, InMemory} from '@zenfs/core'
+import type {AssetMap} from '@devvit/shared-types/Assets.js'
+import * as IDB from 'idb-keyval'
+
+declare global {
+  interface HTMLElementEventMap {}
+  interface HTMLElementTagNameMap {
+    'play-assets': PlayAssets
+  }
+}
+
+export type AssetFilesystemType = 'virtual' | 'local'
+
+const CACHE_LAST_DIR = 'lastMountedDirectory'
+const CACHE_LAST_ZIP = 'lastMountedArchive'
+
+const MIME: {[ext: string]: string} = {
```

Comment:
:no_bell: this might be worth typing as readonly to avoid accidental mutation `{readonly [ext: string]: string}`.